server/memphis_handlers_schemas.go

```diff
@@ -583,41 +441,14 @@ func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
 		}
 	}
 
-	newSchema := models.Schema{
-		ID:   primitive.NewObjectID(),
-		Name: schemaName,
-		Type: schemaType,
-	}
-
-	filter := bson.M{"name": newSchema.Name}
-	update := bson.M{
-		"$setOnInsert": bson.M{
-			"_id":  newSchema.ID,
-			"type": newSchema.Type,
-		},
-	}
-
-	newSchemaVersion := models.SchemaVersion{
-		ID:                primitive.NewObjectID(),
-		VersionNumber:     schemaVersionNumber,
-		Active:            true,
-		CreatedByUser:     user.Username,
-		CreationDate:      time.Now(),
-		SchemaContent:     schemaContent,
-		SchemaId:          newSchema.ID,
-		MessageStructName: messageStructName,
-		Descriptor:        descriptor,
-	}
-
-	opts := options.Update().SetUpsert(true)
-	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
+	newSchema, matchedCount, err := db.UpdateNewSchema(schemaName, schemaType)
```

Comment:
each db function which call update with upsert true should be called upsert instead of update, so please fix it across the entire application